src/crimson/osd/pg_backend.cc

```diff
@@ -900,10 +887,11 @@ PGBackend::cmp_xattr_ierrorator::future<> PGBackend::cmp_xattr(
     switch (osd_op.op.xattr.cmp_mode) {
     case CEPH_OSD_CMPXATTR_MODE_STRING:
       {
-        string val;
-        bp.copy(osd_op.op.xattr.value_len, val);
-        result = do_xattr_cmp_str(osd_op.op.xattr.cmp_op, val, xattr);
-        logger().debug("cmpxattr val={}, xattr= {}", val, xattr);
+        string lhs;
+        bp.copy(osd_op.op.xattr.value_len, lhs);
```

Comment:
`lhs` is a temporary; we don't to extend the life-time. This makes me think about a variant of `copy` that would fill `string_view`.

Reply:
as long as we can ensure that the bufferlist is contiguous. otherwise we pay for the overhead of rebuilding it.